[2.x] Add secret refs, configMap refs and field refs to environment variables

svenvanhees opened the pull request:
Added functionality to add custom environment variables using the valueFrom field, either by using the new addSecretRef function, or settings the value of an env var to an array as described in the updated docs.

Fixes #84

svenvanhees commented:
@rennokki Any ideas as to why the codecov shows a decrease in coverage?

rennokki commented:
It was due to some code duplication. I've updated the methods for secret refs and added new ones for configmap refs and field refs.

svenvanhees commented:
> It was due to some code duplication. I've updated the methods for secret refs and added new ones for configmap refs and field refs.

Ah, seems logical. Saw some code refactors to prevent the duplication. Nice!
